actors/builtin/miner/partition_state.go

```diff
@@ -852,8 +852,12 @@ func (p *Partition) RecordSkippedFaults(
 	}
 	retractedRecoveryPower = PowerForSectors(ssize, retractedRecoverySectors)
 
-	// Ignore skipped faults that are already faults or terminated.
-	newFaults, err := bitfield.SubtractBitField(skipped, p.Terminated)
+	// Ignore skipped faults that are already faults or terminated or recovered
+	newFaults, err := bitfield.SubtractBitField(skipped, retractedRecoveries)
```

Comment:
Hey @zgfzgf this is a redundant operation which does not change behavior. 

The reason is that retractedRecoveries is a subset of p.Recoveries (line 845 above), and p.Recoveries is a subset of p.Faults (see comment here https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/miner/partition_state.go#L28, and validation here: https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/miner/partition_state.go#L950).  And we already subtract p.Faults from newFaults line 860/864 below.

Let me know if you have any questions.  If not please close this PR down.

Reply:
I see. close this RP